handlers.go

```diff
@@ -3602,6 +3606,74 @@ func (s *Server) handleDeleteCollection(c echo.Context, u *User) error {
 	return c.NoContent(http.StatusOK)
 }
 
+type deleteContentFromCollectionBody struct {
+	By    string `json:"by"`
+	Value string `json:"value"`
+}
+
+// handleDeleteContentFromCollection godoc
+// @Summary      Deletes a content from a collection
+// @Description  This endpoint is used to delete an existing content from an existing collection. If two or more files with the same contentid exist in the collection, delete the one in the specified path
+// @Tags         collections
+// @Param        coluuid path string true "Collection ID"
+// @Param        contentid path string true "Content ID"
+// @Param        by body string true "Variable to use when filtering for files (must be either 'path' or 'content_id')"
+// @Param        value body string true "Value of content_id or path to look for"
+// @Produce      json
+// @Success      200  {object}  string
+// @Failure      400  {object}  util.HttpError
+// @Router       /collections/{coluuid}/contents [delete]
+func (s *Server) handleDeleteContentFromCollection(c echo.Context, u *util.User) error {
+	var body deleteContentFromCollectionBody
+	if err := c.Bind(&body); err != nil {
+		return err
+	}
+	coluuid := c.Param("coluuid")
+
+	// check if 'by' is either 'path' or 'content_id'
+	if body.By != "path" && body.By != "content_id" {
+		return &util.HttpError{
+			Code:    http.StatusNotFound,
+			Reason:  util.ERR_INVALID_FILTER,
+			Details: fmt.Sprintf("invalid 'by' value, must be either 'content_id' or 'path', got %s", body.By),
+		}
+	}
+
+	col, err := collections.GetCollection(coluuid, s.DB, u)
+	if err != nil {
+		return err
+	}
+
+	refs := []collections.CollectionRef{}
+	if body.By == "path" {
+		path := body.Value
+		refs, err = collections.GetContentsInPath(coluuid, path, s.DB, u)
+		if err != nil {
+			return err
+		}
+	} else if body.By == "content_id" {
+		contentid := body.Value
+		content, err := util.GetContent(contentid, s.DB, u)
+		if err != nil {
+			return err
+		}
+		if err := s.DB.Model(collections.CollectionRef{}).
+			Where("collection = ?", col.ID).
+			Where("content = ?", content.ID).
+			Scan(&refs).Error; err != nil {
+			return err
+		}
+	} else {
```

Comment:
no need for this block, it will never happen since you did this above `if body.By != "path" && body.By != "content_id" {`

Reply:
I just want to be thorough. In the future should the code change this will let us know a problem is there